tests/reference/cpp-expr3-9c516d4.stdout

```diff
@@ -34,11 +34,11 @@ void test_cast()
     float b;
     a = 2;
     b =   4.20000000000000018e+00;
-    a = (float)(a)*b;
+    a = (int)((float)(a)*b);
     b = b + (float)(1);
     a = 5;
-    a = (float)(a) -   3.89999999999999991e+00;
-    a = (float)(a)/b;
+    a = (int)((float)(a) -   3.89999999999999991e+00);
+    a = (int)((float)(a)/b);
```

Comment:
This is not correct. It needs to return a compiler error. Consider this code:
```python
def test_cast():
    a = 2
    b = 4.2
    a *= b
    b += 1
    a = 5
    print(a, type(a))
    a -= 3.9
    print(a, type(a))
    a /= b
    b = 3/4
    if a < b:
        print("a < b")

test_cast()
```
Which prints in CPython:
```console
$ python tests/expr3.py
(5, <type 'int'>)
(1.1, <type 'float'>)
```
Essentially, the `a -= 3.9` operation changes `a` from an int to a float.

In LPython variables cannot change their types, so we need to give a compiler error for this and not allow it.

The user must explicitly type this.

Reply:
Okay. Well the above code compiles in `main` branch as well. So, you mean any binary expression like `int - float` or `float + complex` shouldn't be allowed right?

Reply:
Or you mean assigning integer to float and float to integer shouldn't be allowed?

Reply:
> In LPython variables cannot change their types, so we need to give a compiler error for this and not allow it.

Well then we will have to remove the casting mechanism all together from LPython, because many test expressions work in `main` branch because casting is performed on `left`, `right` parts of `BinOp`? Should I remove casting and fix the tests and make all the expressions strictly typed?

Reply:
Or specifically the problem is with shorthand operations, `a -= 3.9`? If so then this can be fixed by making `AugAssign` visitor strictly typed.

Reply:
So I assumed that https://github.com/lcompilers/lpython/pull/940#discussion_r942294345 is the case. Since, the tested you posted in https://github.com/lcompilers/lpython/pull/940#discussion_r942279864 is same as the one present in https://github.com/lcompilers/lpython/blob/main/tests/expr3.py. So if we want to raise a compiler error for `a -= 3.9` then we will have to remove the following file https://github.com/lcompilers/lpython/blob/main/tests/reference/asr-expr3-10eb0a4.stdout and replace with `stderr` version.

Reply:
Yes

Reply:
Done. Needed a little bit of fixing at some places. Did that.